HHS/OPRE-OPS
# Ops 1238 delete agreement role restrictions

## What changed

Deleting agreements now is only possible by the created_by, project_officer, or anyone in the team_members list for the agreement. There is a new process for sending a function into the @is_authorized() decorator, which then will check if this function comes back either True or False, as part of the verification if the endpoint is being used correctly. This allows for customization of the is_authorized decorator checks on a per-endpoint granularity.

## Issue

#1238 

## How to test

The pytest unit tests all should come back good, these have been modified to ensure that the user being used for the tests actually is "associated" with the agreement (is one of those three pre-defined entities). Otherwise it would reject and give a 401.

There are gherkin tests in the delete agreement feature that checks for all of this.

## Screenshots

N/A

## Links

N/A
